plugins/evernote.koplugin/JoplinClient.lua

```diff
@@ -0,0 +1,140 @@
+local json = require("json")
+local http = require('socket.http')
+local ltn12 = require('ltn12')
+local socket = require('socket')
+local url = require('socket.url')
+
+local JoplinClient =  {
+    server_ip = "localhost",
+    server_port = 41184,
+    auth_token = ""
+}
+
+function JoplinClient:new(o)
+    o = o or {}
+    self.__index = self
+    setmetatable(o, self)
+    return o
+end
+
+function JoplinClient:_makeRequest(url, method, request_body)
+    local sink = {}
+    local source = {}
+    local request_body = json.encode(request_body)
+    local source = ltn12.source.string(request_body)
+    http.request{
+        url = url,
+        method = method,
+        sink = ltn12.sink.table(sink),
+        source = source,
+        headers = {["Content-Length"] = #request_body,
+            ["Content-Type"] = "application/json"}
+    }
+
+    if not sink[1] then
+        error("No response from Joplin Server") 
+    end
+
+    local response = json.decode(sink[1])
+
+    if response["error"] then
+        error(response["error"])
+    end
+
+    return response
+end
+
+function JoplinClient:ping()
+    local sink = {}
+
+    http.request{
+        url =  "http://"..self.server_ip..":"..self.server_port.."/ping",
+        method = "GET",
+        sink = ltn12.sink.table(sink)
+    }
+
+    if sink[1] == "JoplinClipperServer" then 
+        return true
+    else 
+        return false
+    end
+end
+
+--@return if successful returns id of found note
```

Comment:
I don't think that's valid LDoc style? But regardless, I think it'd be better to write something like this (same for the rest):
```suggestion
--- If successful returns id of found note.
```